client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -253,8 +254,10 @@ public void shutdown(final boolean shutdownFactory) {
             case CREATE_JUST:
                 break;
             case RUNNING:
-                this.mQClientFactory.unregisterProducer(this.defaultMQProducer.getProducerGroup());
                 this.defaultAsyncSenderExecutor.shutdown();
+                ThreadUtils.shutdownGracefully(this.defaultAsyncSenderExecutor,
```

Comment:
Why still invoke  `this.defaultAsyncSenderExecutor.shutdown();` while you already using  `ThreadUtils.shutdownGracefully`?

Reply:
Sorry, forgot to rm this line